src/api/deploy/aws-elastic.ts

```diff
@@ -0,0 +1,267 @@
+          if (!(result && result.data === 0)) {
+            const msg = 'aws configure region failed';
+            observer.next({ type: 'containerError', data: msg});
+            return Promise.reject(-1);
+          }
+
+          // 3. create-application-version
+          let command;
+          if (s3Bucket || codeCommit) {
+            command = {
+              type: CommandType.deploy, command: `aws elasticbeanstalk create-application-version`
+                + ` --application-name "${application}" --version-label "${version}"`
+                + ` --description "${description}" --source-build-information`
+                + ` SourceType="${applicationType}", SourceRepository="${sourceRepository}",`
+                + ` SourceLocation="${sourceLocation}" --auto-create-application`
+            };
+          } else {
+            command = {
+              type: CommandType.deploy, command: `aws elasticbeanstalk create-application-version`
+                + ` --application-name "${application}" --version-label "${version}"`
+                + ` --description "${description}" --auto-create-application`
+            };
+          }
+
+          return attachExec(container, command).toPromise();
+        })
+        .then(() => {
+          // 3. check if environment exists
+          return environmentExists(container, environmentName);
+        })
+        .then(exists => {
+          if (exists) {
+            // 4. create-environment
+            if (environmentTemplate) {
+              let command = {
+                type: CommandType.deploy, command: `aws elasticbeanstalk create-environment`
+                  + ` --application-name "${application}" --environment-name "${environmentName}"`
+                  + ` --template-name "${environmentTemplate}"`
+              };
+
+              return attachExec(container, command)
+                .toPromise()
+                .then(result => {
+                  if (!(result && result.data === 0)) {
+                    const msg = 'aws create environment failed';
+                    observer.next({ type: 'containerError', data: msg});
+                    return Promise.reject(-1);
+                  }
+
+                  return Promise.resolve();
+                });
+            } else if (solutionStackName) {
+              let command = {
+                type: CommandType.deploy, command: `aws elasticbeanstalk create-environment`
+                  + ` --application-name "${application}" --environment-name "${environmentName}"`
+                  + ` --solution-stack-name "${solutionStackName}"`
+              };
+
+              return attachExec(container, command)
+                .toPromise()
+                .then(result => {
+                  if (!(result && result.data === 0)) {
+                    const msg = 'aws create environment failed';
+                    observer.next({ type: 'containerError', data: msg});
+                    return Promise.reject(-1);
```

Comment:
here too, if you are trying to produce an error that would be anything but 0.
0 means success and 1-n means error.

Reply:
I know, what would you suggest instead -1? Like you said, I just want to produce error.

Reply:
you can do `new Error('description')`